packages/babel-helpers/src/helpers.js

```diff
@@ -270,6 +270,151 @@ helpers.createClass = template(`
   })()
 `);
 
+//NOTE: convention is 'descriptor' is used for element descriptors, while 'propertyDescriptor' is used for
+//property descriptor in all helpers related to decorators-2
+helpers.decorate = template(`
+  (function (constructor, undecorated, memberDecorators, heritage) {
+    const prototype = constructor.prototype;
+    let finishers = [];
+    const elementDescriptors = new Map(); // elementDescriptors is meant to be an array, so this will be converted later
+
+    for (const [key, isStatic] of undecorated) {
+      const target = isStatic ? constructor : prototype;
+      const propertyDescriptor = Object.getOwnPropertyDescriptor(target, key);
+      elementDescriptors.set(
+        [key, isStatic], 
+        babelHelpers.makeElementDescriptor(
+          "property",
+          key,
+          isStatic,
+          propertyDescriptor,
+        )
+      );
+    }
+
+    // decorate and store in elementDescriptors
+    for (const [key, decorators, isStatic] of memberDecorators) {
```

Comment:
I wish we could. But if we can't, I'll refactor it to use ES5, it shouldn't be complex. I'll leave this be till the end though, since the code is functionally correct so for the review we could focus on just that